L1Trigger/L1TTrackMatch/interface/L1TkHTMissEmulatorProducer.h

```diff
@@ -0,0 +1,219 @@
+#ifndef L1Trigger_L1TTrackMatch_L1TkHTMissEmulatorProducer_HH
+#define L1Trigger_L1TTrackMatch_L1TkHTMissEmulatorProducer_HH 
+
+// Original Author:  Hardik Routray                                                              
+//         Created:  Mon, 11 Oct 2021 
+
+#include <ap_int.h>
+
+#include <cmath>
+#include <cstdint>
+#include <filesystem>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/L1Trigger/interface/TkJetWord.h"
+
+using namespace std;
+
+// Namespace that defines constants and types used by the HTMiss Emulation
+namespace l1tmhtemu { 
+
+  const unsigned int kPtSize_ = l1t::TkJetWord::TkJetBitWidths::kPtSize;
+  const unsigned int kEtaSize_ = l1t::TkJetWord::TkJetBitWidths::kGlbEtaSize;
+  const unsigned int kPhiSize_ = l1t::TkJetWord::TkJetBitWidths::kGlbPhiSize;
+
+  const unsigned int kInternalPtWidth{kPtSize_};
+  const unsigned int kInternalEtaWidth{kEtaSize_};
+  const unsigned int kInternalPhiWidth{kPhiSize_};
+
+  // extra room for sumPx, sumPy
+  const unsigned int kEtExtra{10};
+
+  const unsigned int kMHTSize{15};  // For output Magnitude default 15
+  const unsigned int kMHTPhiSize{14}; // For output Phi default 14
+  const float kMaxMHT{4096};  // 4 TeV
+  const float kMaxMHTPhi{2 * M_PI};
+
+  typedef ap_uint<5> ntracks_t;
+  typedef ap_uint<kInternalPtWidth> pt_t;
+  typedef ap_int<kInternalEtaWidth> eta_t;
+  typedef ap_int<kInternalPhiWidth> phi_t;
+
+  typedef ap_int<kInternalPtWidth + kEtExtra> Et_t;
+  typedef ap_uint<kMHTSize> MHT_t;
+  typedef ap_uint<kMHTPhiSize> MHTphi_t;
+
+  const unsigned int kMHTBins = 1 << kMHTSize;
+  const unsigned int kMHTPhiBins = 1 << kMHTPhiSize;
+
+  const double kStepPt_ = 0.25;
+  const double kStepEta_ = M_PI / (720);
+  const double kStepPhi_ = M_PI / (720);
+
+  const double kStepPt{kStepPt_}; 
+  const double kStepEta{kStepEta_};
+  const double kStepPhi{kStepPhi_};
+  
+  const double kStepMHT = (l1tmhtemu::kMaxMHT / l1tmhtemu::kMHTBins);
+  const double kStepMHTPhi = (l1tmhtemu::kMaxMHTPhi / l1tmhtemu::kMHTPhiBins);
+
+  const unsigned int kPhiBins = 1 << kInternalPhiWidth;
+
+  const float kMaxCosLUTPhi{M_PI};
+
+  const string kLUTdir{"LUTs/"};
+
+
+  template <typename T>
+    T digitizeSignedValue(double value, unsigned int nBits, double lsb) {
```

Comment:
moving `digitizeSignedValue` to `DataFormats/L1Trigger/interface/EtSum.h` is a good idea. Since only two modules MET and MHT modules use it, I think it is okay for now and can be moved to a centralized tool later.  I can take a look at the MET and MHT modules later and fix this in another PR if required.

Reply:
I think a PR at a later date would be better

Reply:
If you're going to leave this to later, my recommendation would be to create a github issue for this repo so that there is a separate record that this change needs to be made. Then you can link that issue to the second PR so that when you close the second PR, the issue is automatically closed.